src/net.md

```diff
@@ -876,6 +877,61 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-extract-mediawiki-links]: #ex-extract-mediawiki-links
+<a name="ex-extract-mediawiki-links"/>
+## Extract all unique links from a MediaWiki markup
+
+[![reqwest-badge]][reqwest] [![regex-badge]][regex] [![cat-net-badge]][cat-net]
+
+Pull the source of a MediaWiki page using [`reqwest::get`] and then
+look for all entries of internal and external links with
+[`Regex::captures_iter`].
+
+MediaWiki syntax is described [here][MediaWiki link syntax].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate regex;
+
+use std::collections::HashSet;
+use std::io::Read;
+use regex::Regex;
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#         Regex(regex::Error);
+#     }
+# }
+# 
+fn run() -> Result<()> {
+    let mut content = String::new();
+    reqwest::get(
+        "https://en.wikipedia.org/w/index.php?title=Rust_(programming_language)&action=raw",
+    )?
+        .read_to_string(&mut content)?;
+
+    let mut links = Regex::new(r#"\[\[(?P<link>[^\[\]|]*)[^\[\]]*\]\]"#)?
```

Comment:
Also this is what I have so far and I really don't like the if/else there. My iterator-fu may be lacking :/

```rust
    lazy_static! {
        static ref WIKI_REGEX: Regex = 
            Regex::new(r#"\[\[(?P<internal>[^\[\]|]*)[^\[\]]*\]\]|(url=|URL\||\[)(?P<external>http.*?)[ \|}]"#)
            .unwrap();
    }

    let links: HashSet<String> = WIKI_REGEX
        .captures_iter(content.as_str())
        .map(|c| {
            if c.name("internal").is_some() {
                c["internal"].to_lowercase()
            } else {
                // c.name("external").is_some()
                c["external"].to_string()
            }
        })
        .collect();
```